File: android/app/src/main/java/com/carelog/ui/relative/RelativeDashboardViewModel.kt
package com.carelog.ui.relative

import android.util.Log
import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import com.carelog.api.PatientSummary
import com.carelog.api.RelativeApiService
import com.carelog.auth.AuthRepository
import dagger.hilt.android.lifecycle.HiltViewModel
import kotlinx.coroutines.flow.MutableStateFlow
import kotlinx.coroutines.flow.StateFlow
import kotlinx.coroutines.flow.asStateFlow
import kotlinx.coroutines.flow.update
import kotlinx.coroutines.launch
import javax.inject.Inject

/**
 * ViewModel for the Relative Dashboard.
 */
@HiltViewModel
class RelativeDashboardViewModel @Inject constructor(
    private val apiService: RelativeApiService,
    private val authRepository: AuthRepository
) : ViewModel() {

    private val _uiState = MutableStateFlow(RelativeDashboardUiState())
    val uiState: StateFlow<RelativeDashboardUiState> = _uiState.asStateFlow()

    init {
        loadPatientSummary()
    }

    fun refresh() {
        loadPatientSummary()
    }

    private fun loadPatientSummary() {
        viewModelScope.launch {
            _uiState.update { it.copy(isLoading = true, error = null) }

            try {
                // Fetch linked patient ID directly from Cognito (not from cache)
                val patientId = authRepository.fetchLinkedPatientId()
                // Also check currentUser for comparison
                val cachedUser = authRepository.currentUser.value
                val debugInfo = "fetchLinkedPatientId=$patientId, cachedUser.linked=${cachedUser?.linkedPatientId}, cachedUser.email=${cachedUser?.email}"
                Log.d("RelativeDashboard", debugInfo)

                if (patientId == null) {
                    _uiState.update {
                        it.copy(
                            isLoading = false,
                            noPatientRegistered = true
                        )
                    }
                    return@launch
                }

                val summary = apiService.getPatientSummary(patientId)
                _uiState.update {
                    it.copy(
                        isLoading = false,
                        patientSummary = summary,
                        error = null
                    )
                }
            } catch (e: Exception) {
                Log.e("RelativeDashboard", "Error loading patient summary", e)
                _uiState.update {
                    it.copy(
                        isLoading = false,
                        error = "Exception: ${e.javaClass.simpleName}: ${e.message}"
                    )
                }
            }
        }
    }
}

/**
 * UI state for the Relative Dashboard.
 */
data class RelativeDashboardUiState(
    val isLoading: Boolean = false,
    val patientSummary: PatientSummary? = null,
    val noPatientRegistered: Boolean = false,
    val error: String? = null
)

File: ios/CareLog/CareLog/Relative/ViewModels/RelativeDashboardViewModel.swift
import Foundation
import SwiftUI

/// ViewModel for the Relative Dashboard.
@MainActor
class RelativeDashboardViewModel: ObservableObject {
    @Published var isLoading = false
    @Published var patientSummary: PatientSummary?
    @Published var error: String?
    @Published var unreadAlertCount: Int = 0

    private let apiService = RelativeApiService.shared
    private let authService = AuthService.shared

    func loadPatientSummary() async {
        guard !isLoading else { return }

        isLoading = true
        error = nil

        do {
            guard let user = authService.currentUser,
                  let patientId = user.linkedPatientId else {
                error = "No patient linked to this account"
                isLoading = false
                return
            }

            let summary = try await apiService.getPatientSummary(patientId: patientId)
            patientSummary = summary
            unreadAlertCount = summary.unreadAlertCount
        } catch {
            self.error = error.localizedDescription
        }

        isLoading = false
    }

    func refresh() async {
        await loadPatientSummary()
    }
}
